server/api/routes/users.py

```diff
@@ -1,42 +1,86 @@
 from flask import Blueprint, request
 from flask_jwt_extended import create_access_token
 from database import User
+from api.routes.validators.users import valid_email, valid_password
 
 users = Blueprint('users', __name__)
 
-
 @users.route('/login', methods=['POST'])
 def login():
     """POST users/login
 
     Desc: Logs in a user
+          Ensures fields are not blank
+          Validates email used to log in
+          Checks if an account exists at all
+          Checks if an account exists and the password was incorrect
 
     Returns:
         200: success - true
-        401: success - false
+        401: password was incorrect, unauthorized access
+        404: no account was found
     """
     req = request.get_json()
-    user = User.objects(email=req['email'], password=req['password']).first()
-    if user:
-        access_token = create_access_token(identity=user)
+    if req['email'] == '' or req['password'] == '':
+        return {
+            'error': 'Fields must not be empty.'
+        }, 404
+
+    check_email = valid_email(req['email'])
+    if not check_email[0]:
         return {
-            'token': access_token
-        }, 200
-    else:
+            'error': check_email[1]
+        }, 401
+    
+    user = User.objects(email=req['email']).first()
+    if not user:
+        return {
+            'error': 'No account with this email exists.'
+        }, 404
+
+    user = User.objects(email=req['email'], password=req['password']).first()
```

Comment:
You should just be able to check if the user object you have matches the password. That way we're not accessing the database twice

Reply:
How would you differentiate between an account not existing at all and the wrong password being entered

Reply:
So you already query the database using only the username, and it returns the entire user object. If that user object exists, then you're good so far, otherwise return saying the email doesn't have an account. If it does exist, then check if `req['password'] == user.password`. If it does, then the password is correct, otherwise, return saying the incorrect password was entered.

Reply:
Ohh ok that makes sense, I thought you were talking about the second query that used the username and password.